### PR TITLE
Test Coverage and Tests Style improvements

### PR DESCRIPTION
**Pull request by grandizzy**
- RewardsManager:  ajna balance less than rewards tests
- PoolCommons: no interest earned when htp > max price
- ERC721Pool: collateral tests for collection pool
- Pool: loan heap test
- Pool: test move dusty amounts

coverage improved from
```
Summary coverage rate:
  lines......: 99.3% (1524 of 1534 lines)
  functions..: 99.6% (226 of 227 functions)
  branches...: 94.6% (441 of 466 branches)
```
to
```
Summary coverage rate:
  lines......: 99.5% (1527 of 1534 lines)
  functions..: 99.6% (226 of 227 functions)
  branches...: 95.5% (445 of 466 branches)
```

### CODE REVIEW

File: tests/forge/ERC20Pool/ERC20PoolInterestRateAndEMAs.t.sol [hunk 1]
```diff
@@ -746,4 +752,40 @@ contract ERC20PoolInterestRateTestAndEMAs is ERC20HelperContract {
...
+                from:               _borrower3,
+                borrower:           _borrower3,
+                amountToBorrow:     90_000_000_000_000_000 * 1e18,
+                limitIndex:         5000,
+                collateralToPledge: 90_100_000 * 1e18
+            }
+        );
+        skip(100 days);
+
+        assertGt(MAX_PRICE, _htp());
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolInterestRateAndEMAs.t.sol:701]: I think it makes sense to do an `_assertPool` here

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        _mintCollateralAndApproveTokens(_lender1,   1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_lender3,   1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower1, 1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower2, 1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower3, 1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower4, 1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower5, 1_000 * 1e18);
+        _mintCollateralAndApproveTokens(_borrower6, 1_000 * 1e18);
+
+        // Lender 1 adds Quote token accross 2 buckets
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: 3 buckets

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        );
+
+        _assertLoans(
+            {
+                noOfLoans:         6,
+                maxBorrower:       _borrower6,
+                maxThresholdPrice: 6.005769230769230772 * 1e18
+            }
+        );
+
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: comment describing reassignment of head?

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        _assertLoans(
+            {
+                noOfLoans:         6,
+                maxBorrower:       _borrower4,
+                maxThresholdPrice: 14.013461538461538468 * 1e18
+            }
+        );
+
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol:131]: comment here describing reassignment of head

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        _assertLoans(
+            {
+                noOfLoans:         6,
+                maxBorrower:       _borrower6,
+                maxThresholdPrice: 6.005769230769230772 * 1e18
+            }
+        );
+
+        _repayDebt({
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol:155]: Same as above

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+        _assertLoans(
+            {
+                noOfLoans:         6,
+                maxBorrower:       _borrower5,
+                maxThresholdPrice: 5.004807692307692310 * 1e18
+            }
+        );
+
+        _drawDebt(
```
Comment (ith-harvey) [inline tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: same as above

File: tests/forge/RewardsManager.t.sol [hunk 7]
```diff
@@ -1328,6 +1328,29 @@ contract RewardsManagerTest is DSTestPlus {
         assertGt(_ajnaToken.balanceOf(_updater), 0);
 
         // check owner can withdraw the NFT and rewards will be automatically claimed
+
+        uint256 snapshot = vm.snapshot();
+
+        // claimed rewards amount is greater than available tokens in rewards manager contract
```
Comment (ith-harvey) [inline tests/forge/RewardsManager.t.sol:1334]: Is there any code/assertions supporting this claim?

**Review by ith-harvey — CHANGES_REQUESTED** [review]
As per Lucas's changes in his formatting PR I suggest moving forward we clean up all tests from this:
```
_assertLoans(
            {
                noOfLoans:         6,
                maxBorrower:       _borrower6,
                maxThresholdPrice: 12.016346153846153854 * 1e18
            }
        );
```
to this:
```
_assertLoans({
                noOfLoans:         6,
                maxBorrower:       _borrower6,
                maxThresholdPrice: 12.016346153846153854 * 1e18
            });
```
Happy to discuss on standup if you wish too to make sure the whole team is on board.

Additionally can we screenshot the coverage increases or are those automatically visible in the github actions?

File: tests/forge/ERC20Pool/ERC20PoolInterestRateAndEMAs.t.sol (hunk 1 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolInterestRateAndEMAs.t.sol:701]: done in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol (hunk 2 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: changed in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol (hunk 3 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: added in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol (hunk 4 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol:131]: added in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol (hunk 5 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol:155]: added in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol (hunk 6 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/ERC20PoolLoanHeap.t.sol]: added in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

File: tests/forge/RewardsManager.t.sol (hunk 7 above)
Comment (grandizzy) [reply tests/forge/RewardsManager.t.sol:1334]: added in https://github.com/ajna-finance/contracts/pull/541/commits/c7c823fba82ba36416691802ef2b0fc86f0a36b2

**Comment by grandizzy** [comment]
> As per Lucas's changes in his formatting PR I suggest moving forward we clean up all tests from this:
> 
> ```
> _assertLoans(
>             {
>                 noOfLoans:         6,
>                 maxBorrower:       _borrower6,
>                 maxThresholdPrice: 12.016346153846153854 * 1e18
>             }
>         );
> ```
> 
> to this:
> 
> ```
> _assertLoans({
>                 noOfLoans:         6,
>                 maxBorrower:       _borrower6,
>                 maxThresholdPrice: 12.016346153846153854 * 1e18
>             });
> ```
> 
I like the proposal, I applied same style across all unit tests with https://github.com/ajna-finance/contracts/pull/541/commits/3945a5b3556246a76c8f7c0e5898cca75926873f

> Happy to discuss on standup if you wish too to make sure the whole team is on board.
> 
> Additionally can we screenshot the coverage increases or are those automatically visible in the github actions?

I added coverage comparison in PR description

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by MikeHathaway — APPROVED** [review]
LGTM - much cleaner now

**Pull request merged**